src/System.Diagnostics.PerformanceCounter/src/System/Diagnostics/SharedUtils.cs

```diff
@@ -0,0 +1,383 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Security.Permissions;
+using System.Threading;
+using System.Text;
+using Microsoft.Win32;
+using System.ComponentModel;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Diagnostics.CodeAnalysis;
+
+namespace System.Diagnostics
+{
+    internal static class SharedUtils
+    {
+        internal const int UnknownEnvironment = 0;
+        internal const int W2kEnvironment = 1;
+        internal const int NtEnvironment = 2;
+        internal const int NonNtEnvironment = 3;
+        private static volatile int s_environment = UnknownEnvironment;
+        public const int WAIT_OBJECT_0 = 0x00000000;
+        public const int WAIT_ABANDONED = 0x00000080;
+
+        private static Object s_InternalSyncObject;
+        private static Object InternalSyncObject
+        {
+            get
+            {
+                if (s_InternalSyncObject == null)
+                {
+                    Object o = new Object();
+                    Interlocked.CompareExchange(ref s_InternalSyncObject, o, null);
+                }
+                return s_InternalSyncObject;
+            }
+        }
+
+        internal static Win32Exception CreateSafeWin32Exception()
+        {
+            return CreateSafeWin32Exception(0);
+        }
+
+        internal static Win32Exception CreateSafeWin32Exception(int error)
+        {
+            Win32Exception newException = null;
+            try
+            {
+                if (error == 0)
+                    newException = new Win32Exception();
+                else
+                    newException = new Win32Exception(error);
+            }
+            finally
+            {
+            }
+
+            return newException;
+        }
+
+        internal static int CurrentEnvironment
```

Comment:
This will always return `W2kEnvironment` (since it is only included in the version we build for Windows, and .NET Core only supports Win32NT on Windows).
So you can eliminate CurrentEnvironment, CheckEnvironment, CheckNtEnvironment, the mutex, etc.
In fact, I think you can just copy https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.EventLog/src/System/Diagnostics/SharedUtils.cs as @Anipik already had to copy this file and went through this refactoring.

Reply:
And you can remove these strings
```xml
  <data name="WinNTRequired" xml:space="preserve">
    <value>Feature Requires Windows NT</value>
  </data>
  <data name="Win2000Required" xml:space="preserve">
    <value>Feature Requires Windows 2000 or later</value>
  </data>
```